### PR TITLE
NewsSource & NewsItem base class setup

### PR DESCRIPTION
**Pull request by umangutkarsh**
The following tasks are opened in this PR

- [x] Initialize the base class for `NewsSource` which accepts adapter
- [x] Create interface for the adapter.
- [x] Implement `TechCrunchAdapter` adapter.
- [x] Use the adapter with the `NewsSource` class
- [x] Define the the `NewsItem{}` class to store news in a specific format
- [x] Develop the slash command to get news updates on demand (just doing `console.log()` for now)

```
// Issues with this approach
// 1. Typescript circular dependency issue when deploying the app

// Links related to the issue - 
// 1. https://stackoverflow.com/questions/43176006/typeerror-class-extends-value-undefined-is-not-a-function-or-null
// 2. https://stackoverflow.com/questions/37312197/node-js-v6-2-0-class-extends-is-not-a-function-error/37312355#37312355
```
Above issues are addressed with the approach followed

### CODE REVIEW

File: news-aggregation/news-sources/TechCrunchNewsSource.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    ): Promise<any> {}
+
+    async getNews(
+        read: IRead,
+        modify: IModify,
+        room: IRoom,
+        http: IHttp,
+        persis: IPersistence
+    ): Promise<any> {}
+}
```
Comment (abhinavkrin) [inline news-aggregation/news-sources/TechCrunchNewsSource.ts]: This part should be removed. And the `NewsSource` class from `news-aggregation/definitions/NewsSource.ts` should be used.

**Review by abhinavkrin — CHANGES_REQUESTED** [review]

**Comment by abhinavkrin** [comment]
Hey @umangutkarsh 
We can take an alternative approach to avoid the issue you have mentioned,

We can have a single class NewsSource and then have adapter objects for different sources.

For example.
We can have a TechCrunchAdapter 
then initialise a new NewsSource using
```
const newsSource = new NewsSource(new TechCrunchAdapter(...));
or
const newsSource = new NewsSource(TechCrunchAdapter.create(...));
```

**Comment by umangutkarsh** [comment]
> Hey @umangutkarsh We can take an alternative approach to avoid the issue you have mentioned,
> 
> We can have a single class NewsSource and then have adapter objects for different sources.
> 
> For example. We can have a TechCrunchAdapter then initialise a new NewsSource using
> 
> ```
> const newsSource = new NewsSource(new TechCrunchAdapter(...));
> or
> const newsSource = new NewsSource(TechCrunchAdapter.create(...));
> ```

Sure, will create a PR using this. Thanks.

**Comment by umangutkarsh** [comment]
Hi @abhinavkrin . Please review the approach once.

**Review by abhinavkrin — APPROVED** [review]
LGTM! I will approve this PR. I would suggest to use `types` instead of ` any `wherever possible.

**Comment by umangutkarsh** [comment]
> LGTM! I will approve this PR. I would suggest to use `types` instead of `any`wherever possible.

Sure, will make the changes wherever required.

**Pull request merged**